### PR TITLE
LPD-32703 - error returned when using an organization name over the 100 character limit with the headless api is unclear

### PR DESCRIPTION
**Pull request by wanderlast**
Hi @smottal,

https://liferay.atlassian.net/browse/LPD-32703, tied to LPP.

This is a fix related specifically to posting an organization using the API, but has a couple of other fixes related to the issue as well as one due to SF standard changes.

A new exception is added as a separate error condition under the OrganizationNameException for exceeding the max length of an organization name. The previous OrganizationNameException applied to a null check on the name and therefore an additional error condition was created of OrganizationNameException.MustNotBeNull to satisfy the SF rule outlined in [LPD-25246](https://liferay.atlassian.net/browse/LPD-25246) as well as make the usage hopefully more clear. 

OrganizationNameExceptionMapper was kept as-is, but two ExceptionMappers were created per the Headless team's suggestion [here](https://liferay.slack.com/archives/C5E1CRLJY/p1722584773201359?thread_ts=1722472781.281839&cid=C5E1CRLJY). This follows the same pattern as used by the UserAccountScreenNameExceptionMappers.

Notes:
- I did not add an aui:validator on the aui:input in details.jsp due to it being unnecessary there -- the auI:input already adds a maxlength attribute of 100 to the field and therefore the issue cannot happen there. This is also why the existing OrganizationNameException aui:error was not changed since I don't believe we can trigger the MustNotExceedMaximumLength exception in this file. I'd be happy to add it if you feel it's necessary though due to a need I didn't anticipate for it.

### CODE REVIEW

**Comment by wanderlast** [comment]
ci:test:sf

**Comment by wanderlast** [comment]
ci:test:sf

File: portal-kernel/src/com/liferay/portal/kernel/exception/OrganizationNameException.java [hunk 1]
```diff
@@ -5,24 +5,35 @@
...
  * @author Brian Wing Shun Chan
  */
 public class OrganizationNameException extends PortalException {
 
-	public OrganizationNameException() {
-	}
+	public static class MustNotBeNull extends OrganizationNameException {
+
+		public MustNotBeNull() {
+			super("Organization name must not be null.");
```
Comment (smottal) [inline portal-kernel/src/com/liferay/portal/kernel/exception/OrganizationNameException.java:18]: Organization name must not be null. -> Name must not be null

File: portal-kernel/src/com/liferay/portal/kernel/exception/OrganizationNameException.java [hunk 2]
```diff
@@ -5,24 +5,35 @@
...
 
-	public OrganizationNameException(String msg, Throwable throwable) {
-		super(msg, throwable);
+	public static class MustNotExceedMaximumLength
+		extends OrganizationNameException {
+
+		public MustNotExceedMaximumLength(String name, int nameMaxLength) {
+			super(
+				StringBundler.concat(
+					"Organization name", name, " must have fewer than ",
```
Comment (smottal) [inline portal-kernel/src/com/liferay/portal/kernel/exception/OrganizationNameException.java:29]: Remove Organization

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by wanderlast** [comment]
ci:test:sf

File: portal-kernel/src/com/liferay/portal/kernel/exception/OrganizationNameException.java [hunk 3]
```diff
@@ -5,24 +5,35 @@
...
 
-	public OrganizationNameException(String msg, Throwable throwable) {
-		super(msg, throwable);
+	public static class MustNotExceedMaximumLength
+		extends OrganizationNameException {
+
+		public MustNotExceedMaximumLength(String name, int nameMaxLength) {
+			super(
+				StringBundler.concat(
+					"Name", name, " must have fewer than ", nameMaxLength,
```
Comment (smottal) [inline portal-kernel/src/com/liferay/portal/kernel/exception/OrganizationNameException.java:29]: sorry @wanderlast , I think it's missing a space in the string "Name"

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by wanderlast** [comment]
ci:test:sf

**Comment by wanderlast** [comment]
ci:test:relevant

**Review by smottal — APPROVED** [review]

**Comment by smottal** [comment]
ci:test:relevant

**Pull request closed without merging**